i18n-util/test/spec/UrlUtil_spec.js

```diff
@@ -20,12 +19,76 @@ describe('UrlUtil.getAliasUrl', () => {
 });
 
 describe('UrlUtil.getRedirectUrl', () => {
-  test('generates redirect url correctly', () => {
+  const oldWindowLocation = window.location
+  const setup = (hash) => {
+    // reset values
+    delete window.location;
+    window.location = {
+      hash
+    };
+  };
+  afterEach(() => {
+    window.location = oldWindowLocation;
+  });
+  test('generates redirect url correctly for admin docs with locale query param', () => {
+    setup('cshid=Settings_Emails_SMS?locale=ja');
     const result = getRedirectUrl({
       locale: 'ja',
       aliasUrl: 'help.foo.com/okta_help.htm#cshid=Settings_Emails_SMS?locale=ja'
     });
     expect(result).toEqual('help.foo.com/ja/prod/okta_help.htm#cshid=Settings_Emails_SMS');
   });
 
+  test('generates redirect url correctly for admin docs without locale query param', () => {
+    setup('cshid=Settings_Emails_SMS');
+    const result = getRedirectUrl({
+      locale: 'en',
+      aliasUrl: 'help.foo.com/okta_help.htm#cshid=Settings_Emails_SMS'
+    });
+    expect(result).toEqual('help.foo.com/en/prod/okta_help.htm#cshid=Settings_Emails_SMS');
+  });
+
+  test('generates redirect url correctly for end-user docs with locale query param', () => {
+    setup('cshid=Settings_Emails_SMS?locale=ja&type=end-user');
+    const result = getRedirectUrl({
+      locale: 'ja-jp',
+      aliasUrl: 'help.foo.com/okta_help.htm#cshid=Settings_Emails_SMS?locale=ja&type=end-user'
+    });
+    expect(result).toEqual('help.foo.com/end-user/ja-jp/okta_help.htm#cshid=Settings_Emails_SMS');
+  });
+
+  test('generates redirect url correctly for end-user docs without locale query param', () => {
+    setup('cshid=Settings_Emails_SMS?type=end-user');
+    const result = getRedirectUrl({
+      locale: 'ja-jp',
+      aliasUrl: 'help.foo.com/okta_help.htm#cshid=Settings_Emails_SMS?type=end-user'
+    });
+    expect(result).toEqual('help.foo.com/end-user/ja-jp/okta_help.htm#cshid=Settings_Emails_SMS');
+  });
+
+});
+
+describe('UrlUtil.getUrlParameter', () => {
+  const oldWindowLocation = window.location
+  const setup = (hash) => {
+    // reset values
+    delete window.location;
+    window.location = {
+      hash
+    };
+  };
```

Comment:
This is an anti-pattern for mocking global objects. Instead, you can rely on mocks:

```js
let windowSpy = jest.spyOn(window, 'window', 'get');
windowSpy.mockImplementation(() => ({
  location: {
    hash,
  },
});
```